app/models/create_cap_proposal.rb

```diff
@@ -0,0 +1,90 @@
+class CreateCapProposal
+  def initialize(auction)
+    if !auction.is_a?(Presenter::Auction)
+      fail "an instance of Auction::Presenter must be passed to CreateCapProposal::initialize"
```

Comment:
it seems like we shouldn't have to type check like this in our `initialize` methods